wordpress.org/public_html/wp-content/plugins/plugin-directory/class-plugin-search.php

```diff
@@ -0,0 +1,404 @@
+		}
+
+
+		// Some extra fields here
+		if ( isset( $es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][0][ 'multi_match' ] ) ) {
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][0][ 'multi_match' ][ 'boost' ] = 2;
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][0][ 'multi_match' ][ 'fields' ] = ( $this->is_english ? [
+				0 => 'title_en',
+				1 => 'excerpt_en',
+				2 => 'description_en^1',
+				3 => 'taxonomy.plugin_tags.name',
+			] : [
+				'title_' . $this->locale,
+				'excerpt_' . $this->locale,
+				'description_' . $this->locale . '^' . $this->desc_boost,
+				'title_en^' . $this->en_boost,
+				'excerpt_en^' . $this->en_boost,
+				'description_en^' . $this->desc_en_boost,
+				'taxonomy.plugin_tags.name',
+			] );
+		}
+
+		// And some more fancy bits here
+		if ( isset( $es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ] ) && 1 === count( $es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ] ) ) {
+			$search_phrase = $es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][0][ 'multi_match' ][ 'query' ];
+
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][] = [
+				'multi_match' => [
+				'query' => $search_phrase,
+				'fields' => ( $this->is_english ? [
+					0 => 'title_en.ngram',
+				] : [
+					'title_' . $this->locale . '.ngram',
+					'title_en.ngram^' . $this->en_boost,
+				] ),
+				'type' => 'phrase',
+				'boost' => 0.2,
+				],
+			];
+
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][] = [
+				'multi_match' => [
+				  'query' => $search_phrase,
+				  'fields' => ( $this->is_english ? [
+					0 => 'title_en',
+					1 => 'slug_text',
+				  ] : [
+					'title_' . $this->locale,
+					'title_en^' . $this->en_boost,
+					'slug_text',
+				  ] ),
+				  'type' => 'best_fields',
+				  'boost' => 2,
+				],
+			];
+
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][] = [
+				'multi_match' => [
+				  'query' => $search_phrase,
+				  'fields' => ( $this->is_english ? [
+					0 => 'excerpt_en',
+					1 => 'description_en^1',
+					2 => 'taxonomy.plugin_tags.name',
+				  ] : [
+					'excerpt_' . $this->locale,
+					'description_' . $this->locale . '^' . $this->desc_boost,
+					'excerpt_en^' . $this->en_boost,
+					'description_en^' . $this->desc_en_boost,
+					'taxonomy.plugin_tags.name',
+				  ] ),
+				  'type' => 'best_fields',
+				  'boost' => 2,
+				],
+			];
+
+			$es_query_args[ 'query' ][ 'function_score' ][ 'query' ][ 'bool' ][ 'should' ][] = [
+				'multi_match' => [
+				  'query' => $search_phrase,
+				  'fields' => [
+					0 => 'author',
+					1 => 'contributors',
+				  ],
+				  'type' => 'best_fields',
+				  'boost' => 2,
+				],
+			];
+		}
+
+		if ( isset( $es_query_args[ 'query' ][ 'function_score' ][ 'functions' ] ) ) {
```

Comment:
The query builder supports all of this https://github.com/Automattic/jetpack/blob/2138f8787efc1e3552410666dfd14bae14b42c1a/projects/plugins/jetpack/_inc/lib/jetpack-wpes-query-builder/jetpack-wpes-query-builder.php#L87